app/utils/ethereum/ethereum.js

```diff
@@ -0,0 +1,232 @@
+  };
+
+  chainCheck = async () => {
+    if (parseInt(this.connectedChain.id, 16) !== Number(process.env.CHAIN_ID)) {
+      await this.setChain({
+        chainId: `0x${Number(process.env.CHAIN_ID).toString(16)}`,
+      });
+    }
+  };
+
+  walletLogIn = async (previouslyConnectedWallet) => {
+    try {
+      document.getElementsByTagName('body')[0].style.position = 'fixed';
+    } catch (err) {}
+
+    if (previouslyConnectedWallet) {
+      await this.connect({ autoSelect: previouslyConnectedWallet });
+    } else {
+      await this.connect();
+    }
+
+    if (!this.connectedWallets?.length) {
+      document.getElementsByTagName('body')[0].style.position = 'relative';
+      deleteCookie('connectedWallet');
+      return;
+    }
+
+    await this.chainCheck();
+
+    this.provider = new ethers.providers.Web3Provider(this.wallet.provider);
+    const signer = await this.provider.getSigner();
+    this.contractUser = new Contract(process.env.USER_ADDRESS, PeeranhaUser, signer);
+    this.contractCommunity = new Contract(process.env.COMMUNITY_ADDRESS, PeeranhaCommunity, signer);
+    this.contractContent = new Contract(process.env.CONTENT_ADDRESS, PeeranhaContent, signer);
+    this.contractToken = new Contract(process.env.PEERANHA_TOKEN, PeeranhaToken, this.provider);
+
+    document.getElementsByTagName('body')[0].style.position = 'relative';
+
+    return this.selectedAccount.toLowerCase();
+  };
+
+  resetWalletState = async () => {
+    await this.disconnect(this.wallet);
+    deleteCookie('connectedWallet');
+    this.selectedAccount = null;
+  };
+
+  setSelectedAccount = (account) => {
+    this.selectedAccount = account?.toLowerCase();
+  };
+
+  getSelectedAccount = () => this.selectedAccount;
+
+  getProfile = async (userAddress) => {
+    const user = await this.getUserDataWithArgs(GET_USER_BY_ADDRESS, [userAddress]);
+
+    return {
+      creationTime: user.creationTime,
+      ipfsDoc: user.ipfsDoc,
+      rating: user.rating,
+      followedCommunities: user.followedCommunities,
+      ipfsHash: getIpfsHashFromBytes32(user.ipfsDoc.hash),
+    };
+  };
+
+  waitForCloseModal() {
+    return new Promise((resolve) => {
+      this.stopWaiting = function () {
+        resolve();
+      };
+    });
+  }
+
+  getUserDataWithArgs = async (action, args) => await this.contractUserReads[action](...args);
+
+  getCommunityDataWithArgs = async (action, args) =>
+    await this.contractCommunityReads[action](...args);
+
+  getContentDataWithArgs = async (action, args) => await this.contractContentReads[action](...args);
+
+  getTokenDataWithArgs = async (action, args) => await this.contractTokenReads[action](...args);
+
+  getCommunityFromContract = async (id) => {
+    const rawCommunity = await this.getCommunityDataWithArgs(GET_COMMUNITY, [id]);
+    const communityInfo = JSON.parse(
+      await getText(getIpfsHashFromBytes32(rawCommunity.ipfsDoc.hash)),
+    );
+    return {
+      id: +id,
```

Comment:
may be better to use `id: Number(id)` ?

Reply:
https://github.com/peeranha/peeranha-web/pull/1402/commits/78897b69b4ad523ef680321471acfef0cbeb7f4f